static/main.js

```diff
@@ -51,6 +51,7 @@ var SimpleCook = require('./simplecook').SimpleCook;
 var HistoryWidget = require('./history-widget').HistoryWidget;
 var History = require('./history');
 var presentation = require('./presentation');
+require('./tests/_all');
```

Comment:
Does this mean the tests are pulled into the main site?

Reply:
Yes, the code is pulled in, but not run.

I fiddled around with conditional inclusion in the PUG, but not sure that would work because of the webpack/ts requirements.

But since it's always included, it also means you can notice that the tests need changing when you change an interface or type, that's a big plus in maintaining them and also developing new frontend classes that can be unit-tested.

Reply:
It might be possible to have a JS global set by webpack that controls whether or not the tests get required.

Then we can make sure it's not set when doing the final webpack for production.

Reply:
@mattgodbolt the typescript files I added are independent of cypress, and cypress, while a devDependency, doesn't actually run anything unless you actually execute `npx cypress run`.

Reply:
I am now realizing actually maybe I missed something, because I am using assert from mocha in the typescript test, which doesn't make sense to have in production.

Let's hold off on merging this until I figure this out.

Reply:
confirmed the webpack define-plugin works, but it's a bit scary as the linter will need some proof it exists, but in fact it will never exist, webpack will just either emit or leave out the code within the if statement